packages/actor-init-query/lib/HttpServiceSparqlEndpoint.ts

```diff
@@ -550,13 +550,14 @@ export class HttpServiceSparqlEndpoint {
       return;
     }
 
-    const s = request.url;
+    const protocol = request.socket && 'encrypted' in request.socket && request.socket.encrypted ? 'https' : 'http';
```

Comment:
I'm sorry I didn't notice this before, but it should actually be safe to just use relative URLs, and not include the base URL, as RDF parsers can determine these on their own.
But I don't remember how the rest of this logic here was implemented, so you definitely need to double-check this.
You can do this by setting up a comunica sparql endpoint locally, and then querying over that endpoint (either via the comunica cli, or using https://query.comunica.dev/).